In [1]:
import pandas as pd
import numpy as np
import os
import gensim
import nltk

In [2]:
nltk.download('all')

[nltk_data] Downloading collection 'all'
[nltk_data]    | 
[nltk_data]    | Downloading package abc to
[nltk_data]    |     C:\Users\Sunny\AppData\Roaming\nltk_data...
[nltk_data]    |   Package abc is already up-to-date!
[nltk_data]    | Downloading package alpino to
[nltk_data]    |     C:\Users\Sunny\AppData\Roaming\nltk_data...
[nltk_data]    |   Package alpino is already up-to-date!
[nltk_data]    | Downloading package averaged_perceptron_tagger to
[nltk_data]    |     C:\Users\Sunny\AppData\Roaming\nltk_data...
[nltk_data]    |   Package averaged_perceptron_tagger is already up-
[nltk_data]    |       to-date!
[nltk_data]    | Downloading package averaged_perceptron_tagger_eng to
[nltk_data]    |     C:\Users\Sunny\AppData\Roaming\nltk_data...
[nltk_data]    |   Package averaged_perceptron_tagger_eng is already
[nltk_data]    |       up-to-date!
[nltk_data]    | Downloading package averaged_perceptron_tagger_ru to
[nltk_data]    |     C:\Users\Sunny\AppData\Roaming\nltk_data...
[

True

In [3]:
from nltk import sent_tokenize
from gensim.utils import simple_preprocess

In [4]:
for file in os.listdir('./data/'):
    print(file)

story1.txt
story2.txt


In [20]:
f = open(r"D:\complete_content_new\Full-Stack-GenAI-Bootcamp-1.0\Class-06-07-11-12-Apr-Word2vec-with-practical\data\story1.txt",encoding='unicode_escape')

In [21]:
f

<_io.TextIOWrapper name='D:\\complete_content_new\\Full-Stack-GenAI-Bootcamp-1.0\\Class-06-07-11-12-Apr-Word2vec-with-practical\\data\\story1.txt' mode='r' encoding='unicode_escape'>

In [22]:
corpus =f.read()

In [23]:
corpus

'Once upon a time, there was a young boy named Arjun who lived in a small town surrounded by hills and rivers. He was very curious about how things worked and always asked questions. His parents often smiled at his curiosity and encouraged him to learn more.\n\nEvery day after school, Arjun would sit with his books and explore different topics. He was especially fascinated by science and technology. He wanted to understand how machines think and whether they could become as intelligent as humans.\n\nOne day, while browsing through an old computer at school, Arjun discovered the concept of artificial intelligence. He read about how machines can learn from data, recognize patterns, and even make decisions. This idea completely amazed him.\n\nHe started imagining a future where machines could help humans solve complex problems. That day, he made a decision that he wanted to become an AI engineer.\n\nArjun began learning programming using free resources. At first, everything seemed difficu

In [24]:
sent_tokenize(corpus)

['Once upon a time, there was a young boy named Arjun who lived in a small town surrounded by hills and rivers.',
 'He was very curious about how things worked and always asked questions.',
 'His parents often smiled at his curiosity and encouraged him to learn more.',
 'Every day after school, Arjun would sit with his books and explore different topics.',
 'He was especially fascinated by science and technology.',
 'He wanted to understand how machines think and whether they could become as intelligent as humans.',
 'One day, while browsing through an old computer at school, Arjun discovered the concept of artificial intelligence.',
 'He read about how machines can learn from data, recognize patterns, and even make decisions.',
 'This idea completely amazed him.',
 'He started imagining a future where machines could help humans solve complex problems.',
 'That day, he made a decision that he wanted to become an AI engineer.',
 'Arjun began learning programming using free resources.',


In [25]:
raw_sent = sent_tokenize(corpus)

In [26]:
for i in raw_sent:
  print(simple_preprocess(i))

['once', 'upon', 'time', 'there', 'was', 'young', 'boy', 'named', 'arjun', 'who', 'lived', 'in', 'small', 'town', 'surrounded', 'by', 'hills', 'and', 'rivers']
['he', 'was', 'very', 'curious', 'about', 'how', 'things', 'worked', 'and', 'always', 'asked', 'questions']
['his', 'parents', 'often', 'smiled', 'at', 'his', 'curiosity', 'and', 'encouraged', 'him', 'to', 'learn', 'more']
['every', 'day', 'after', 'school', 'arjun', 'would', 'sit', 'with', 'his', 'books', 'and', 'explore', 'different', 'topics']
['he', 'was', 'especially', 'fascinated', 'by', 'science', 'and', 'technology']
['he', 'wanted', 'to', 'understand', 'how', 'machines', 'think', 'and', 'whether', 'they', 'could', 'become', 'as', 'intelligent', 'as', 'humans']
['one', 'day', 'while', 'browsing', 'through', 'an', 'old', 'computer', 'at', 'school', 'arjun', 'discovered', 'the', 'concept', 'of', 'artificial', 'intelligence']
['he', 'read', 'about', 'how', 'machines', 'can', 'learn', 'from', 'data', 'recognize', 'patterns',

In [27]:
folder_path = r"D:\complete_content_new\Full-Stack-GenAI-Bootcamp-1.0\Class-06-07-11-12-Apr-Word2vec-with-practical\data"
story = []
for filename in os.listdir(folder_path):
  file_path=os.path.join(folder_path,filename)
  with open(file_path,encoding='unicode_escape') as f:
    corpus=f.read()
  raw_sent=sent_tokenize(corpus)
  for sent in raw_sent:
    story.append(simple_preprocess(sent))

In [28]:
story

[['once',
  'upon',
  'time',
  'there',
  'was',
  'young',
  'boy',
  'named',
  'arjun',
  'who',
  'lived',
  'in',
  'small',
  'town',
  'surrounded',
  'by',
  'hills',
  'and',
  'rivers'],
 ['he',
  'was',
  'very',
  'curious',
  'about',
  'how',
  'things',
  'worked',
  'and',
  'always',
  'asked',
  'questions'],
 ['his',
  'parents',
  'often',
  'smiled',
  'at',
  'his',
  'curiosity',
  'and',
  'encouraged',
  'him',
  'to',
  'learn',
  'more'],
 ['every',
  'day',
  'after',
  'school',
  'arjun',
  'would',
  'sit',
  'with',
  'his',
  'books',
  'and',
  'explore',
  'different',
  'topics'],
 ['he',
  'was',
  'especially',
  'fascinated',
  'by',
  'science',
  'and',
  'technology'],
 ['he',
  'wanted',
  'to',
  'understand',
  'how',
  'machines',
  'think',
  'and',
  'whether',
  'they',
  'could',
  'become',
  'as',
  'intelligent',
  'as',
  'humans'],
 ['one',
  'day',
  'while',
  'browsing',
  'through',
  'an',
  'old',
  'computer',
  'at',
  'sc

In [29]:
story[0]

['once',
 'upon',
 'time',
 'there',
 'was',
 'young',
 'boy',
 'named',
 'arjun',
 'who',
 'lived',
 'in',
 'small',
 'town',
 'surrounded',
 'by',
 'hills',
 'and',
 'rivers']

In [30]:
story[1]

['he',
 'was',
 'very',
 'curious',
 'about',
 'how',
 'things',
 'worked',
 'and',
 'always',
 'asked',
 'questions']

In [ ]:
sunny->100
Divyansh->6
Rahul->3

In [ ]:
# Word2Vec(..., sg=0)  # CBOW #by default
# Word2Vec(..., sg=1)  # Skip-gram

In [ ]:
custom_model =gensim.models.Word2Vec(
    window=10,
    min_count=5,
    vector_size=150
    
)

## these all the value are the hyperparameter

In [32]:
custom_model.build_vocab(story)

In [34]:
custom_model.corpus_count

44

In [35]:
custom_model.train(story, total_examples=custom_model.corpus_count, epochs=5)

(119, 2610)

(119, 2610)
Part	Meaning
119	number of documents
2610	number of unique words

In [36]:
custom_model.wv["arjun"]

array([ 2.6837648e-03,  2.8979357e-03,  6.6388524e-03, -2.9810630e-03,
       -9.2678453e-04, -4.8770588e-03, -6.4641684e-03, -6.0503520e-03,
       -6.8084226e-04, -4.3351529e-03,  3.2344444e-03, -4.1086348e-03,
        1.6795450e-03,  4.9268454e-04, -2.2612980e-03, -6.5211818e-04,
        6.6558407e-03,  6.0951482e-03, -2.9747835e-03,  6.0557672e-03,
       -3.7611320e-03,  3.9526452e-03, -2.0619777e-03,  2.2907313e-03,
        2.0105771e-03,  4.6007000e-03, -1.5832029e-03,  5.8508962e-03,
        5.0607324e-03, -6.3660997e-03, -5.3393990e-03, -5.0912779e-03,
        1.9495630e-03, -1.8639804e-03, -4.6199737e-03, -5.4182662e-03,
        5.5407137e-03,  1.3234081e-03, -6.2196818e-03, -3.1952166e-03,
        2.0923335e-03, -3.1411920e-03,  3.5213172e-03, -2.8242120e-03,
        1.7607786e-03, -5.3630467e-03,  4.1403784e-03,  3.2111451e-03,
        5.2263401e-04,  2.0095946e-03, -5.8171460e-03,  1.4217742e-03,
       -5.8630249e-04, -6.2126359e-03, -6.2853270e-03, -9.4161421e-04,
      

In [37]:
len(custom_model.wv["arjun"])

150